### DISCUSSION TITLE
limit concurrent connections for one Channel (WebSockets)

### DISCUSSION THREAD

**Discussion by rawezhcode** [vania-dart/framework#96] [Q&A]
What is the limit for one channel?

I have a Q&A app that starts live at a certain time at night.
For example, the game begins at 10 pm with more than 10k users.
I'm currently having trouble in real-time,
my app that is made with (flutter),
I only use (broadcast) to send questions to users.
What VPS is good for a centrifuge accepting 10k concurrent connections in one Channel?

**Comment by javad-zobeidi** [comment]
The minimum system requirements to handle 10k concurrent connections in one channel are:

RAM: At least 4GB
CPU: 4 CPU cores 
Storage: NVMe SSD for fast read/write operations, especially important for high-performance environments and reducing latency
Network: A fast and stable network connection with at least 1Gbps bandwidth, since real-time apps with many concurrent users require good network throughput.

For a production-grade environment with 10k users on a single channel, it’s crucial to monitor resource usage and scale the infrastructure horizontally if needed (e.g., deploying multiple instances of Centrifugo or using load balancing).

Additionally, it would help if you also considered the following optimizations:

Ensure the VPS has adequate support for WebSocket connections and low-latency networking.

Vertical scaling: If you're running on a single VPS, make sure it's capable of handling the peak load with sufficient overhead. Consider vertical scaling if necessary before moving to a multi-instance setup.

Horizontal scaling: To handle more users and ensure scalability, you might need a load balancer to distribute connections across multiple VPS or cloud instances.
